src/Datasource/EntityTrait.php

```diff
@@ -787,6 +787,38 @@ public function getDirty()
     }
 
     /**
+     * return value if property exists otherwise throw Exception
+     *
+     * @param string $property The field to get value for.
+     * @return mixed
+     */
+    public function getOrFail($property)
+    {
+        $value = $this->get($property);
+        if (!$this->propertyExists($property) && $value === null) {
+            throw new \BadMethodCallException('Property "' . $property . '" not found.');
```

Comment:
This is not a good exception. BadMethodCallException: `Exception thrown if a callback refers to an undefined method or if some arguments are missing. `
http://php.net/manual/de/class.badmethodcallexception.php

This is what you want - I think: http://php.net/manual/de/class.unexpectedvalueexception.php